library/core/src/main/java/com/google/android/exoplayer2/source/SimpleMediaSource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.source;
+
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.Timeline;
+
+/**
+ * Single-child {@link CompositeMediaSource}.
+ */
+public abstract class SimpleMediaSource extends CompositeMediaSource<Void> {
+
+  /**
+   * @deprecated - Use {@link #onChildSourceInfoRefreshed}.
+   */
+  @Deprecated
+  @Override
+  protected final void onChildSourceInfoRefreshed(
+      Void id, MediaSource mediaSource, Timeline timeline
+  ) {
+    onChildSourceInfoRefreshed(mediaSource, timeline);
+  }
+
+  /**
+   * Called when the source info has been refreshed.
+   *
+   * @param mediaSource The child source whose source info has been refreshed.
+   * @param timeline The timeline of the source.
+   */
+  protected abstract void onChildSourceInfoRefreshed(
+      MediaSource mediaSource, Timeline timeline
+  );
+
+  /**
+   * Prepares a source.
+   *
+   * <p>{@link #onChildSourceInfoRefreshed(MediaSource, Timeline)} will be called when the
+   * child source updates its timeline with the same {@code id} passed to this method.
+   *
+   * <p>If sources aren't explicitly released with {@link #releaseChildSource()} they will be
+   * released in {@link #releaseSourceInternal()}.
+   *
+   * @param mediaSource The child {@link MediaSource}.
+   */
+  protected final void prepareChildSource(MediaSource mediaSource) {
+    prepareChildSource(CHILD_SOURCE_ID, mediaSource);
+  }
+
+  /**
+   * Enables the child source.
+   */
+  protected final void enableChildSource() {
+    enableChildSource(CHILD_SOURCE_ID);
+  }
+
+  /**
+   * Disables the child source.
+   */
+  protected final void disableChildSource() {
+    disableChildSource(CHILD_SOURCE_ID);
+  }
+
+  /**
+   * Releases the child source.
+   */
+  protected final void releaseChildSource() {
+    releaseChildSource(CHILD_SOURCE_ID);
+  }
+
+  /**
+   * @deprecated - Use {@link #getWindowIndexForChildWindowIndex(int)}.
```

Comment:
This method isn't really deprecated and you can just remove the comment entirely. Same for the other cases below.

Reply:
Why not deprecate it? Would receivers of this class really want a method that takes an `id`?